faker/providers/ssn/es_MX/__init__.py

```diff
@@ -0,0 +1,250 @@
+# coding=utf-8
+from __future__ import unicode_literals
+
+import random
+import string
+
+from .. import Provider as BaseProvider
+
+
+ALPHABET = string.ascii_uppercase
+ALPHANUMERIC = string.digits + ALPHABET
+VOWELS = "AEIOU"
+CONSONANTS = [
+    letter
+    for letter in ALPHABET
+    if letter not in VOWELS
+]
+
+# https://es.wikipedia.org/wiki/Plantilla:Abreviaciones_de_los_estados_de_M%C3%A9xico
+STATES_RENAPO = [
+    "AS",
+    "BC",
+    "BS",
+    "CC",
+    "CS",
+    "CH",
+    "DF",
+    "CL",
+    "CM",
+    "DG",
+    "GT",
+    "GR",
+    "HG",
+    "JC",
+    "MC",
+    "MN",
+    "MS",
+    "NT",
+    "NL",
+    "OC",
+    "PL",
+    "QO",
+    "QR",
+    "SP",
+    "SL",
+    "SR",
+    "TC",
+    "TS",
+    "TL",
+    "VZ",
+    "YN",
+    "ZS",
+    "NE",  # Foreign Born
+]
+
+FORBIDDEN_WORDS = {
+    "BUEI": "BUEX",
+    "BUEY": "BUEX",
+    "CACA": "CACX",
+    "CACO": "CACX",
+    "CAGA": "CAGX",
+    "CAGO": "CAGX",
+    "CAKA": "CAKX",
+    "CAKO": "CAKX",
+    "COGE": "COGX",
+    "COJA": "COJX",
+    "COJE": "COJX",
+    "COJI": "COJX",
+    "COJO": "COJX",
+    "CULO": "CULX",
+    "FETO": "FETX",
+    "GUEY": "GUEX",
+    "JOTO": "JOTX",
+    "KACA": "KACX",
+    "KACO": "KACX",
+    "KAGA": "KAGX",
+    "KAGO": "KAGX",
+    "KOGE": "KOGX",
+    "KOJO": "KOJX",
+    "KAKA": "KAKX",
+    "KULO": "KULX",
+    "MAME": "MAMX",
+    "MAMO": "MAMX",
+    "MEAR": "MEAX",
+    "MEAS": "MEAX",
+    "MEON": "MEOX",
+    "MION": "MIOX",
+    "MOCO": "MOCX",
+    "MULA": "MULX",
+    "PEDA": "PEDX",
+    "PEDO": "PEDX",
+    "PENE": "PENX",
+    "PUTA": "PUTX",
+    "PUTO": "PUTX",
+    "QULO": "QULX",
+    "RATA": "RATX",
+    "RUIN": "RUIN",
+}
+
+CURP_CHARACTERS = "0123456789ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
+
+
+def _reduce_digits(n):
+    """
+    Sum of digits of a number until sum becomes single digit.
+
+    Example:
+        658 => 6 + 5 + 8 = 19 => 1 + 9 = 10 => 1
+    """
+    if n == 0:
+        return 0
+    if n % 9 == 0:
+        return 9
+    else:
```

Comment:
Pylint does not approve the `if-else-return`. Not sure it is important.

Reply:
doesn't cost much to make pylint happy, let's fix it

Reply:
pylint should be happier now :)